File: hw7/src/Stack.java
import java.util.ArrayList;

public class Stack {
  private ArrayList<Pos> cells;

  public Stack() {
    cells = new ArrayList<>();
  }

  public void push(Pos cell) {
    cells.add(cell);
  }


  Pos pop() {
    return cells.remove(cells.size()-1);
  }

  public boolean exist(Pos p) {
    return cells.contains(p);
  }

  Pos peek() {
    return cells.get(cells.size()-1);
  }

  boolean empty() {
    return cells.isEmpty();
  }


  public int size() {
    return cells.size();
  }
  public Pos get(int i) {
    return cells.get(i);
  }
}

File: hw3/src/Stack.cpp
#include "Stack.h"

void Stack::push(const Pos &cell) {
    cells.push_back(cell);
}

Pos Stack::pop() {
    Pos p = cells.back();
    cells.pop_back();
    return p;
}

bool Stack::exist(const Pos &p) {
    for (auto &cell : cells)
        if (cell == p)
            return true;
    return false;
}

Pos Stack::peek() {
    return cells.back();
}

bool Stack::empty() {
    return (cells.empty());
}

Stack::Stack() : cells(std::vector<Pos>()) {}
